### DISCUSSION TITLE
After 0.5.8 connection reboots every few seconds

### DISCUSSION THREAD

**Discussion by Gnec220** [lbbrhzn/ocpp#1230] [Q&A]
Hello,
I have a wallbox pulsar plus. After the update of integreation to 0.5.9 the connection to wallbox is broken. I get the following error in my logs.

can some one help me ?

`This error originated from a custom integration.

Logger: custom_components.ocpp
Source: custom_components/ocpp/api.py:1023
integration: Open Charge Point Protocol (OCPP) (documentation, issues)
First occurred: 12:42:30 AM (494 occurrences)
Last logged: 3:25:03 PM

Unexpected exception in connection to '131883': 'TypeConstraintViolationError: Payload for Action is syntactically correct but at least one of the fields violates data type constraints (e.g. “somestring”: 12), {'cause': "'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long", 'ocpp_message': <CallResult - unique_id=866237ec-4e0c-4624-91be-0b7ef97558de, action=GetConfiguration, payload={'unknownKey': [], 'configurationKey': [{'readonly': False, 'key': 'MeterValuesSampledData', 'value': 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'}]}>}'
Unexpected exception in connection to '131883': 'TypeConstraintViolationError: Payload for Action is syntactically correct but at least one of the fields violates data type constraints (e.g. “somestring”: 12), {'cause': "'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long", 'ocpp_message': <CallResult - unique_id=33a38ff8-6249-490d-a7e0-21ef9c1e590e, action=GetConfiguration, payload={'unknownKey': [], 'configurationKey': [{'readonly': False, 'key': 'MeterValuesSampledData', 'value': 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'}]}>}'
Unexpected exception in connection to '131883': 'TypeConstraintViolationError: Payload for Action is syntactically correct but at least one of the fields violates data type constraints (e.g. “somestring”: 12), {'cause': "'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long", 'ocpp_message': <CallResult - unique_id=35a1cfb6-5efc-4cd2-9d3e-0423b71d4b1c, action=GetConfiguration, payload={'unknownKey': [], 'configurationKey': [{'readonly': False, 'key': 'MeterValuesSampledData', 'value': 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'}]}>}'
Unexpected exception in connection to '131883': 'TypeConstraintViolationError: Payload for Action is syntactically correct but at least one of the fields violates data type constraints (e.g. “somestring”: 12), {'cause': "'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long", 'ocpp_message': <CallResult - unique_id=372bc87e-2a91-4a8c-90e5-911a83ee78d5, action=GetConfiguration, payload={'unknownKey': [], 'configurationKey': [{'readonly': False, 'key': 'MeterValuesSampledData', 'value': 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'}]}>}'
Unexpected exception in connection to '131883': 'TypeConstraintViolationError: Payload for Action is syntactically correct but at least one of the fields violates data type constraints (e.g. “somestring”: 12), {'cause': "'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long", 'ocpp_message': <CallResult - unique_id=1bcdb970-2464-4a41-9f88-7cfed94b7c49, action=GetConfiguration, payload={'unknownKey': [], 'configurationKey': [{'readonly': False, 'key': 'MeterValuesSampledData', 'value': 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'}]}>}'
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/ocpp/messages.py", line 226, in validate_payload
    validator.validate(message.payload)
  File "/usr/local/lib/python3.12/site-packages/jsonschema/validators.py", line 451, in validate
    raise error
jsonschema.exceptions.ValidationError: 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long

Failed validating 'maxLength' in schema['properties']['configurationKey']['items']['properties']['value']:
    {'maxLength': 500, 'type': 'string'}

On instance['configurationKey'][0]['value']:
    'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/ocpp/api.py", line 1023, in run
    await asyncio.gather(*self.tasks)
  File "/config/custom_components/ocpp/api.py", line 468, in post_connect
    chgr_measurands = await self.get_configuration(key)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ocpp/api.py", line 890, in get_configuration
    resp = await self.call(req)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/ocpp/charge_point.py", line 422, in call
    validate_payload(response, self._ocpp_version)
  File "/usr/local/lib/python3.12/site-packages/ocpp/messages.py", line 240, in validate_payload
    raise TypeConstraintViolationError(
ocpp.exceptions.TypeConstraintViolationError: TypeConstraintViolationError: Payload for Action is syntactically correct but at least one of the fields violates data type constraints (e.g. “somestring”: 12),  {'cause': "'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N' is too long", 'ocpp_message': <CallResult - unique_id=8eac53d8-692d-45b3-9ebf-58550aff4f6f, action=GetConfiguration, payload={'unknownKey': [], 'configurationKey': [{'readonly': False, 'key': 'MeterValuesSampledData', 'value': 'Current.Export.L1,Current.Export.L2,Current.Export.L3,Current.Import.L1,Current.Import.L2,Current.Import.L3,Current.Offered,Energy.Active.Export.Interval,Energy.Active.Export.Register,Energy.Active.Import.Interval,Energy.Active.Import.Register,Energy.Reactive.Export.Interval,Energy.Reactive.Export.Register,Energy.Reactive.Import.Interval,Energy.Reactive.Import.Register,Frequency,Power.Active.Export,Power.Active.Import,Power.Factor,Power.Offered,Power.Reactive.Export,Power.Reactive.Import,RPM,SoC,Temperature,Voltage.L1-N,Voltage.L2-N,Voltage.L3-N'}]}>}`

**Comment by drc38** [comment]
Are you using the Skip OCPP schema validation option?

**Reply by Gnec220** [reply]
Yes i am using the skip ocpp schema validation option.

**Reply by drc38** [reply]
Try using the OCPP Configure service to set fewer measurands.

**Reply by Gnec220** [reply]
I tried that too. For example in the measuranda I choose only temperature. But the outcome itself didn't change.

**Reply by drc38** [reply]
See https://github.com/lbbrhzn/ocpp/issues/1203 for downgrading to a version where you can select measurands on setup

**Reply by drc38** [reply]
Should be resolved in v0.5.12 as you can select measurands manually
